pkg/image/registry/imagesignature/rest.go

```diff
@@ -31,12 +31,19 @@ func (r *REST) New() runtime.Object {
 	return &imageapi.ImageSignature{}
 }
 
-func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, _ bool) (runtime.Object, error) {
+func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, _ bool) (runtime.Object, error) {
 	signature := obj.(*imageapi.ImageSignature)
 
 	if err := rest.BeforeCreate(Strategy, ctx, obj); err != nil {
 		return nil, err
 	}
+	// at this point we have a fully formed object.  It is time to call the validators that the apiserver
+	// handling chain wants to enforce.
+	if createValidation != nil {
+		if err := createValidation(obj.DeepCopyObject()); err != nil {
```

Comment:
@sttts the DeepCopyObject can be nil, right? do we care here?

Reply:
> @sttts the DeepCopyObject can be nil, right? do we care here?

We don't care here

Reply:
DeepCopyObject is an interface func. If obj is nil, this call panics. But DeepCopyObject will never return nil. Note this is different for DeepCopy() as it is called on a struct pointer.

Reply:
But I agree with @deads2k: here obj is non-nil.